src/classes/PS_CommitmentRequest.cls

```diff
@@ -138,13 +159,20 @@ public inherited sharing class PS_CommitmentRequest {
 
     /***
     * @description Constructs the Commitment request body for the specified Recurring Donation
-    * @param recordId Recurring Donation Id
+    * @param rd Recurring Donation record
     * @param token Payment Method Token
     * @return RequestBody
     */
-    public RequestBody getRequestBody(Id recordId, String token) {
-        npe03__Recurring_Donation__c rd = getRecurringDonation(recordId);
 
+    public RequestBody getRequestBody(npe03__Recurring_Donation__c rd, String token) {
+        Boolean isCreateCommitment = String.isBlank(rd.CommitmentId__c);
+
+        return isCreateCommitment
```

Comment:
When building a commitment request, use commitment Id to differentiate between create (POST) and edit (PATCH) request body content. Do not use the `recordId` since an existing RD can be changed so its commitment Id is added on the form (custom fields) so it is commitment edit, or a credit card added for existing RD but it is a create commitment request not edit. (W-8554656)